spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -898,9 +901,8 @@ unit_hook_id: { driver->enableHookIDMode(); }
 unit_hook_attribute
               : FOREACH                          { $$ = builder->attribute("foreach", __loc__); }
```

Comment:
I am wondering whether we should as a follow-up add support for `%foreach` using the existing `foreach` machinery and deprecate hook identifier. This would make things more uniform since all hooks but `foreach` start with `%...`.

Reply:
Had the same thought but would indeed rather not do as part of this.